src/services/external/excalidraw-backend/excalidraw.redis.server.factory.provider.ts

```diff
@@ -0,0 +1,322 @@
+      authService,
+      authorizationService,
+      whiteboardRtService
+    ),
+};
+
+const factory = async (
+  appId: string,
+  logger: LoggerService,
+  configService: ConfigService,
+  authService: AuthenticationService,
+  authorizationService: AuthorizationService,
+  whiteboardRtService: WhiteboardRtService
+) => {
+  const port = configService.get(ConfigurationTypes.HOSTING).whiteboard_rt.port;
+
+  if (!port) {
+    logger.error('Port not provided!', EXCALIDRAW_SERVER);
+    return;
+  }
+
+  const kratosPublicBaseUrl = configService.get(ConfigurationTypes.IDENTITY)
+    .authentication.providers.ory.kratos_public_base_url_server;
+
+  const kratosClient = new FrontendApi(
+    new Configuration({
+      basePath: kratosPublicBaseUrl,
+    })
+  );
+
+  const httpServer = http.createServer();
+
+  httpServer.listen(port, () => {
+    logger?.verbose?.(
+      `listening on port: ${port}`,
+      LogContext.EXCALIDRAW_SERVER
+    );
+  });
+
+  try {
+    const { host, port } = configService.get(ConfigurationTypes.STORAGE).redis;
+
+    const pubClient = createClient({ url: `redis://${host}:${port}` });
+    const subClient = pubClient.duplicate();
+
+    const redisAdapter = createAdapter(pubClient, subClient);
+
+    const wsServer = new SocketIO(httpServer, {
+      transports: ['websocket'],
+      allowEIO3: true,
+      adapter: redisAdapter,
+    });
+
+    wsServer.on(CONNECTION, async socket => {
+      const agentInfo = await getUserInfo(
+        kratosClient,
+        socket.handshake.headers,
+        logger,
+        authService
+      );
+
+      if (!agentInfo) {
+        closeConnection(socket, 'Not able to authenticate user');
+        return;
+      }
+
+      logger?.verbose?.(
+        `User '${agentInfo.userID}' established connection`,
+        LogContext.EXCALIDRAW_SERVER
+      );
+
+      wsServer.to(socket.id).emit(INIT_ROOM);
+      // client events ONLY
+      socket.on(
+        JOIN_ROOM,
+        async (roomID: string) =>
+          await joinRoomEventHandler(
+            roomID,
+            agentInfo,
+            socket,
+            wsServer,
+            whiteboardRtService,
+            authorizationService,
+            logger
+          )
+      );
+
+      socket.on(SERVER_BROADCAST, (roomID: string, data: ArrayBuffer) =>
+        serverBroadcastEventHandler(roomID, data, socket)
+      );
+
+      socket.on(
+        SERVER_VOLATILE_BROADCAST,
+        (roomID: string, data: ArrayBuffer) =>
+          serverVolatileBroadcastEventHandler(roomID, data, socket)
+      );
+
+      socket.on(
+        DISCONNECTING,
+        async () =>
+          await disconnectingEventHandler(agentInfo, wsServer, socket, logger)
+      );
+
+      socket.on(DISCONNECT, () => disconnectEventHandler(socket));
+    });
+  } catch (error) {
+    logger.error(error, LogContext.EXCALIDRAW_SERVER);
+  }
+};
+
+/* Sets the user into the context field or closes the connection */
+const authenticate = async (
```

Comment:
`authenticate` and `getUserInfo` are util functions, split them in separate files.